### PR TITLE
Create staging_workshop.yml

### PR DESCRIPTION
**Pull request by pbking**
Adding a deploy action on changes to /trunk for https://twentytwentytwo-trunk.mystagingwebsite.com/

**Description**

This deploys the content of the repo to the /themes folder in the noted site.  The expectation is that this would be the "workshop" where there will be many examples of different types of content and edge conditions.

The "starting content" is the [theme-test-data](https://github.com/WPTT/theme-test-data).

A separate site (likely 2022.wordpress.net) should also be setup to be a more "public face" of the theme with content better designed to demonstrate the strengths of the theme and properly achieve the design targets.

`STAGING_USER` and `STAGING_PASSWORD` values will need to be stored as secrets in order for this action to work.

Addresses #21 (though I still think 2022.wordpress.net would benefit from an automated deployment alongside this)

### CODE REVIEW

File: .github/workflows/staging_workshop.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: TwentyTwentyTwo Staging Workshop Deploy
```
Comment (jffng) [inline .github/workflows/staging_workshop.yml]:
```suggestion
name: Deploy to Test Site
```

**Review by jffng — APPROVED** [review]
I added the respective secrets to the repository, so we should be all set.

Left one small comment about the workflow name, thank you for setting this up @pbking !
